Code/GraphMol/Substruct/SubstructUtils.cpp

```diff
@@ -55,11 +55,19 @@ bool bondCompat(const BOND_SPTR &b1, const BOND_SPTR &b2,
   PRECONDITION(b2, "bad bond");
   bool res;
   if (useQueryQueryMatches && b1->hasQuery() && b2->hasQuery()) {
-    res = static_cast<QueryBond *>(b1.get())
-              ->QueryMatch(static_cast<QueryBond *>(b2.get()));
+    res = static_cast<QueryBond *>(b1.get())->QueryMatch(
+        static_cast<QueryBond *>(b2.get()));
   } else {
     res = b1->Match(b2);
   }
+  if (res && b1->getBondType() == Bond::DATIVE &&
```

Comment:
Generally mismatched bond types lead to the bonds not matching, so res would already be false and we wouldn't get here.
In the event that the query is generic enough to match the dative bond without being a dative bond itself, I think it's probably safest to assume a match without checking the direction.